lib/modules/manager/haskell-cabal/extract.spec.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  countPackageNameLength,
+  countPrecedingIndentation,
+  extractNamesAndRanges,
+  findExtents,
+  splitSingleDependency,
+} from './extract';
+
+describe('modules/manager/haskell-cabal/extract', () => {
+  describe('countPackageNameLength', () => {
+    it.each`
+      input    | expected
+      ${'-'}   | ${null}
+      ${'-j'}  | ${null}
+      ${'-H'}  | ${null}
+      ${'j-'}  | ${null}
+      ${'3-'}  | ${null}
+      ${'-3'}  | ${null}
+      ${'3'}   | ${null}
+      ${'æ'}   | ${null}
+      ${'æe'}  | ${null}
+      ${'j'}   | ${1}
+      ${'H'}   | ${1}
+      ${'0ad'} | ${3}
+      ${'3d'}  | ${2}
+    `('matches $input', ({ input, expected }) => {
+      const maybeIndex = countPackageNameLength(input);
+      expect(maybeIndex).toStrictEqual(expected);
+    });
+  });
+
+  describe('countPrecedingIndentation()', () => {
+    it.each`
+      content                                       | index | expected
+      ${'\tbuild-depends: base\n\tother-field: hi'} | ${1}  | ${1}
+      ${' build-depends: base'}                     | ${1}  | ${1}
+      ${'a\tb'}                                     | ${0}  | ${0}
+      ${'a\tb'}                                     | ${2}  | ${1}
+      ${'a b'}                                      | ${2}  | ${1}
+      ${'  b'}                                      | ${2}  | ${2}
+    `(
+      'countPrecedingIndentation($content, $index)',
+      ({ content, index, expected }) => {
+        expect(countPrecedingIndentation(content, index)).toBe(expected);
+      },
+    );
+  });
+
+  describe('findExtents()', () => {
+    it.each`
+      content                | indent | expected
+      ${'a: b\n\tc: d'}      | ${1}   | ${10}
+      ${'a: b'}              | ${2}   | ${4}
+      ${'a: b\n\tc: d'}      | ${2}   | ${4}
+      ${'a: b\n '}           | ${2}   | ${6}
+      ${'a: b\n c: d\ne: f'} | ${1}   | ${10}
+    `('findExtents($indent, $content)', ({ indent, content, expected }) => {
+      expect(findExtents(indent, content)).toBe(expected);
+    });
+  });
+
+  describe('splitSingleDependency()', () => {
+    it.each`
+      depLine              | expectedName | expectedRange
+      ${'base >=2 && <3'}  | ${'base'}    | ${'>=2 && <3'}
+      ${'base >=2 && <3 '} | ${'base'}    | ${'>=2 && <3'}
+      ${'base>=2&&<3'}     | ${'base'}    | ${'>=2&&<3'}
+      ${'base'}            | ${'base'}    | ${''}
+    `(
+      'splitSingleDependency($depLine)',
+      ({ depLine, expectedName, expectedRange }) => {
+        const res = splitSingleDependency(depLine);
+        expect(res?.name).toBe(expectedName);
+        expect(res?.range).toBe(expectedRange);
+      },
+    );
+  });
+
+  describe('extractNamesAndRanges()', () => {
+    it('trims replaceString', () => {
+      const res = extractNamesAndRanges(' a , b ');
+      expect(res).toHaveLength(2);
```

Comment:
Use rather `toEqual` to get a better assertion on a single line

Reply:
Looks not resolved to me.

Reply:
I must be misunderstanding something. Currently, there is a single call to toEqual. It needs multiple lines since I think it would be cumbersome to map over the array and extract just the replaceString. Would it be possible to detail the changes you are imagining?

Note that the code snippet above is outdated, there is no longer a call to toHaveLength.